### PR TITLE
Issue/244 usort enter scope

### LINKED ISSUE #244: UsortEnterScope
Reference: https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/usort.cairo#L15

Depends on issue #245

### PR DESCRIPTION
**Pull request by pxyxyrus**
Fix #244

### CODE REVIEW

**Comment by pxyxyrus** [comment]
I haven't added any unit tests for this case, and I believe this functionality will need to be checked using an integration test once everything related to usort is implemented.

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,4 @@ cpu.out
 .DS_Store
 .idea/
 vendor/
+.venv
```
Comment (MaksymMalicki) [inline .gitignore:13]: idk if this is generated by python vm compilation or some other script, but we don't use any python in this repo so try to remove it or compile cairo in some other dir / local repo

File: pkg/hintrunner/zero/zerohint_usort.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import (
+	"github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/hinter"
+	VM "github.com/NethermindEth/cairo-vm-go/pkg/vm"
+)
+
+func newUsortEnterScopeHinter() hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "UsortEnterScope",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			__usort_max_size, err := ctx.ScopeManager.GetVariableValue("__usort_max_size")
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_usort.go:12]: i think we should stick to camel case

File: pkg/hintrunner/zero/zerohint_usort.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+			ctx.ScopeManager.EnterScope(map[string]any{
+				"__usort_max_size": __usort_max_size,
+			})
+
+			return nil
+		},
+	}
+}
+
+func createUsortEnterScopeHinter(resolver hintReferenceResolver) (hinter.Hinter, error) {
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_usort.go:26]: i think you can remove resolver from args if you dont use it

File: pkg/hintrunner/zero/zerohint_usort_test.go [hunk 4]
```diff
@@ -0,0 +1 @@
+package zero
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_usort_test.go:1]: regarding the [comment](https://github.com/NethermindEth/cairo-vm-go/pull/339#issuecomment-2040713274) in the issue, you could enter scope and add the variable from globals in the ctxInit method in the unittest. Later you can check in the check method, if the value in given scope is the same which you have added.

Comment (pxyxyrus) [reply pkg/hintrunner/zero/zerohint_usort_test.go:1]: Added tests

File: pkg/hintrunner/zero/zerohint_usort.go (hunk 3 above)
Comment (pxyxyrus) [reply pkg/hintrunner/zero/zerohint_usort.go]: removed

File: .gitignore (hunk 1 above)
Comment (pxyxyrus) [reply .gitignore]: Sorry, it was my own setting and I didn't mean to commit this, but now it's removed.

File: pkg/hintrunner/zero/zerohint_usort_test.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+	"github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/hinter"
+)
+
+func TestZeroHintUsort(t *testing.T) {
+	runHinterTests(t, map[string][]hintTestCase{
+		"UsortEnterScope": {
+			{
+				ctxInit: func(ctx *hinter.HintRunnerContext) {
+					ctx.ScopeManager.EnterScope(map[string]any{
```
Comment (cicr99) [inline pkg/hintrunner/zero/zerohint_usort_test.go:14]: I would rather use here `InitializeScopeManager` function instead

File: .gitignore [hunk 6]
```diff
@@ -9,4 +9,4 @@ cpu.out
 .python-version
 .DS_Store
 .idea/
-vendor/
+vendor/
```
Comment (cicr99) [inline .gitignore:12]: No need to commit this change

**Review by cicr99 — APPROVED** [review]

**Pull request merged**